### ISSUE TITLE
Collaboration ideas with Satpy, Polar2Grid, and Geo2Grid

### ISSUE THREAD

**Issue by djhoese** [NRLMMD-GEOIPS/geoips#310]
As you may know the CSPP LEO Polar2Grid and CSPP Geo Geo2Grid project depend on Satpy. GeoIPS has some use of Satpy and plans for other integrations. I had a thought while watching the workshop today: I should make GeoIPS plugins loadable by Satpy...or at the very least in Polar2Grid/Geo2Grid. Besides a conversion to dask, much of the data-related things should be pretty easy to convert between. At the worst points it might require writing some Satpy-compatible YAML for "enhancements" and colormaps, but even then, not that bad.

I think this would really bring us full circle where GeoIPS and Satpy are able to depend on each other and provide code to each other. I still haven't dived into the GeoIPS internals, but it could provide another debug avenue for plugin developers.

Just an idea that came to mind. Not sure if I'll ever have time to act on it, but it might be a good excuse for some low-level collaboration.

**Comment by jsolbrig** [comment]
I like this idea. We would definitely need to think about how best to implement this and whether it requires some changes to GeoIPS and/or Satpy to achieve. We should definitely continue this conversation.

Along these lines, I wonder if we should co-host the next GeoIPS and Satpy workshops so that we can have some more productive conversations on how to move forward with something like this.

**Comment by djhoese** [comment]
Yeah the more I'm learning about GeoIPS I think the biggest concern and difficulty all of us have is using tools like xarray (or just python dicts in general) to refer to things that have more than one identifying feature. Xarray Datasets versus DataArrays versus the newer DataTree project and encoding things in strings (ex. "MED:solar_zenith_angle") versus Satpy's approach of a complex `DataID` object with custom identifier properties (ex. `DataID(name="C01", calibration="reflectance", resolution=1000, wavelength=WavelengthRange(low_wl, nominal_wl, high_wl))`).
